Libs/Optimize/ParticleSystem/itkParticleGradientDescentPositionOptimizer.txx

```diff
@@ -185,6 +188,12 @@ namespace itk
               {
                 m_TimeSteps[dom][k] *= factor;
                 if (gradmag > maxchange) maxchange = gradmag;
+
+                // debugg
+                std::stringstream stream;
+                stream << "###########################################################################\n" << dom << " " << " " << k << " " << oldpoint << " " << newpoint << std::endl;
+                if(!m_ParticleSystem->GetDomain(dom)->GetConstraints()->IsAnyViolated(oldpoint) && m_ParticleSystem->GetDomain(dom)->GetConstraints()->IsAnyViolated(newpoint) ) std::cerr << stream.str();
```

Comment:
we shouldn't have this and any other cout/cerr statements merged in. they clutter the terminal instead of just showing the optimizer progress.